modules/substrate/src/lib.rs

```diff
@@ -222,18 +232,19 @@ impl<T: Trait> Module<T> {
 		}
 	}
 
-	/// Return the latest header which enacts an authority set change
-	/// and still needs a finality proof.
+	/// Returns a list of headers which require finality proofs.
 	///
-	/// Will return None if there are no headers which are missing finality proofs.
-	pub fn requires_justification() -> Option<BridgedHeader<T>> {
+	/// These headers require proofs because they enact authority set changes.
+	pub fn require_justifications() -> Vec<BridgedHeader<T>> {
```

Comment:
Probably also worth to change to:
```rust
pub fn require_justifications() -> Vec<(BridgedBlockNumber<T>, BridgedBlockHash<T>)> {}
```
? Or do you think some clients will need **headers**, not the **headers ids**? This isn't critical in any way, though

Reply:
I think returning HeaderIds is fine. This will also save us some trips to storage